great_expectations/datasource/fluent/spark_datasource.py

```diff
@@ -99,17 +161,17 @@ def test_connection(self, test_assets: bool = True) -> None:
         Raises:
             TestConnectionError: If the connection test fails.
         """
-        raise NotImplementedError(
-            """One needs to implement "test_connection" on a _SparkDatasource subclass."""
-        )
+        try:
+            self.get_spark()
+        except ConnectionError as e:
+            raise TestConnectionError(e) from e
```

Comment:
TIL (or maybe Today I Remembered) that we weren't testing spark connections on the standard SparkDatasource.
😬